ledger/accountdb.go

```diff
@@ -360,8 +738,16 @@ func writeCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, bals []norm
 		return err
 	}
 
+	var insertRscStmt *sql.Stmt
+	insertRscStmt, err = tx.PrepareContext(ctx, "INSERT INTO catchpointresources(addrid, aidx, data) VALUES(?, ?, ?)")
+	if err != nil {
+		return err
+	}
+
+	var result sql.Result
+	var rowID int64
 	for _, balance := range bals {
-		result, err := insertAcctStmt.ExecContext(ctx, balance.address[:], balance.normalizedBalance, balance.encodedAccountData)
+		result, err = insertAcctStmt.ExecContext(ctx, balance.address[:], balance.normalizedBalance, balance.encodedAccountData)
```

Comment:
The trigger for that was the variable shadowing, which we try to avoid. In this case, it's not obvious that the `err` variable here is the *same* one as defined above.
Second motivation is to avoid unneeded variable cleanup on each iteration of the loop. These variable are always being written to, and therefore doesn't require to be re-initialized ( admittedly, the compiler might be able to optimize for that.. in some cases ).